encode_datasets.py

```diff
@@ -2059,7 +2064,228 @@ def process(inputfile, outfile):
+    fmeta = open(f"{dataset}/{kind}_meta.txt", "w+")
+    df=pd.read_json("/content/"+file)
+
+    if kind=='test':
+        questions=df[['Question','Tweet','qid']].values
+    else:
+        questions=df[['Question','Tweet','qid','Answer']].values
+
+    for row in range(len(questions)):
+        id=questions[row][2]
+        contexts=questions[row][1].strip().replace("\n", "").replace("\t", "").replace("   ", " ").replace("  ", " ")
+        question=questions[row][0].strip().replace("\n", "").replace("\t", "").replace("   ", " ").replace("  ", " ")
+        if '?' not in question:
+            question = question + "?"
+        fmeta.write(f"{id}\n")
+        if kind=='test':
+            answer_string="-"
+        else:
+            answer_string=questions[row][3][0].strip().replace("\n", "").replace("\t", "").replace("   ", " ").replace("  ", " ")
+        fout.write(f"{question} \\n {contexts}\t{answer_string} \n")
+    return len(questions) 
+
+def tweetqa():
+    train_count=tweetqa_process("train.json","tweetqa","train")
+    dev_count=tweetqa_process("dev.json","tweetqa","dev")
+    test_count=tweetqa_process("test.json","tweetqa","test")
+    with open(f"/content/tweetqa/counts.json", "w+") as outfile:
+        json.dump({"train": train_count, "dev": dev_count, "test": test_count}, outfile)
+
+#measuring massive multitask language understanding dataset
+def mmmlu_process(dataset, kind):
+    fout = open(f"{dataset}/{kind}.tsv", "w+")
+    fmeta = open(f"{dataset}/{kind}_meta.txt", "w+")
+    counter=0
+
+    file_path="/content/"+kind+"/*.csv"
+    for file in glob.glob(file_path):
+        df=pd.read_csv(file,header=None)
+        questions=df.values
+
+        for row in range(len(questions)):
+            counter+=1
+            question=questions[row][0].strip().rstrip("\n").replace("\n", "").replace("\t", "").replace("   ", " ").replace("  ", " ")
+            option1=" (A) "+questions[row][1].strip().replace("\n", "").replace("\t", "").replace("   ", " ").replace("  ", " ")
+            option2=" (B) "+questions[row][2].strip().replace("\n", "").replace("\t", "").replace("   ", " ").replace("  ", " ")
+            option3=" (C) "+questions[row][3].strip().replace("\n", "").replace("\t", "").replace("   ", " ").replace("  ", " ")
+            option4=" (D) "+questions[row][4].strip().replace("\n", "").replace("\t", "").replace("   ", " ").replace("  ", " ")
+            options = option1 + option2 + option3 + option4 
+
+            answer=questions[row][5]
+            answer_index = ord(answer)-ord('A')
+            answer_string=questions[row][answer_index+1].strip().replace("\n", "").replace("\t", "").replace("   ", " ").replace("  ", " ")
+            fmeta.write(f"{counter}\t{answer} \n")
+            fout.write(f"{question} \\n{options} \t{answer_string} \n")
+    return counter
+
+def mmmlu():
+    val_count=mmmlu_process("measuring_massive_multitask_language_understanding","val")
```

Comment:
"val" should be changed to "dev".

Reply:
In this dataset, "val" and "dev" are two distinct splits. Should I merge them? As we don't have train split, we can change "val" to "train".